src/mcts/utils.h

```diff
@@ -237,11 +303,32 @@ class MctsResult {
       }
     }
     maxWeight = weight(maxWeight);
+
+    // Print the piVals of all possible actions
+    // std::cout << "PiVals of all possible actions:" << std::endl;
+    // for (size_t i = 0; i != mctsPolicy.size(); ++i) {
+    //   std::cout << "Action " << i << ": " << mctsPolicy[i] << std::endl;
+    // }
+  
     bestAction = sampleDiscreteProbability(
         mctsPolicy.size(), maxWeight,
         [&](size_t i) { return weight(mctsPolicy[i]); }, *rng_);
   }
 
+  // ChatGPT
+  // assume already normalized
+  // void sample() {
```

Comment:
Why is this commented function here?

I also advise against using the Mersenne Twister, it's [slow and has randomness problems](https://arxiv.org/abs/1910.06437).

Xoshiro PRNGs are a decent alternative.
https://quuxplusone.github.io/blog/2021/11/23/xoshiro/

Reply:
Even Mersenne Twister has problems!!!?!?!

Reply:
Good to know, this can be an "issue"